app/recipe/serializers.py

```diff
@@ -0,0 +1,46 @@
+from rest_framework import serializers
+from .models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ingredient
+        # fields = "__all__"
+        fields = ("name",)
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    ingredients = IngredientSerializer(read_only=False, many=True)
+
+    class Meta:
+        model = Recipe
+        fields = (
+            "id",
+            "name",
+            "description",
+            "ingredients",
+        )
+        read_only_fields = [
+            "id",
+        ]
+
+    def create(self, validated_data):
+        recipe = Recipe.objects.create(
+            name=validated_data["name"], description=validated_data["description"]
+        )
+        ingredients_data = validated_data["ingredients"]
+
+        if ingredients_data:
+            for ingredient in ingredients_data:
+                Ingredient.objects.create(recipe=recipe, name=ingredient["name"])
+        return recipe
+
+    def update(self, instance, validated_data):
+        recipe = instance
```

Comment:
Hmm are you updating the name or description if it is provided? I believe you are only updating the ingredients.

Reply:
Yes! I was only updating the ingredients. I just fixed it :)